src/main/java/seedu/address/ui/CalendarPanel.java

```diff
@@ -0,0 +1,341 @@
+        int daysInMonth = DAYS_IN_MONTH[m];
+        if (gc.isLeapYear(gc.get(Calendar.YEAR)) && m == 1) {
+            daysInMonth++;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (int i = daysInMonth; i > 0; i--) {
+            sb.append(i).append(" ");
+        }
+        String[] allDays = sb.toString().split("\\s");
+        String[] neededDays = Arrays.copyOfRange(allDays, 0, leadGap);
+        neededDays = reverse(neededDays);
+        return neededDays;
+    }
+
+    /**
+     * Reverses a String array.
+     * @param s String array to be reversed
+     * @return Reversed String array.
+     */
+    private String[] reverse(String[] s) {
+        for (int i = 0; i < (s.length - (i + 1)); i++) {
+            String temp = s[s.length - (i + 1)];
+            s[s.length - (i + 1)] = s[i];
+            s[i] = temp;
+        }
+
+        return s;
+    }
+
+    /**
+     * Generates array containing days from selected month.
+     * @param numDays Number of days in selected month
+     * @return String array containing days from selected month.
+     */
+    private String[] currMonth(int numDays) {
+        String[] result = new String[numDays];
+        for (int i = 0; i < numDays; i++) {
+            result[i] = Integer.toString(i + 1);
+        }
+        return result;
+    }
+
+    /**
+     * Generates array containing days from next month.
+     * @param currMonth Current selected month
+     * @param tailGap Number of days needed from next month
+     * @return String array containing days from next month.
+     */
+    private String[] nextMonth(int currMonth, int tailGap) {
+        Calendar c = Calendar.getInstance();
+        c.set(Calendar.YEAR, year);
+        c.set(Calendar.MONTH, currMonth);
+        c.add(Calendar.MONTH, 1);
+        int y = c.get(Calendar.YEAR);
+        int m = c.get(Calendar.MONTH);
+
+        GregorianCalendar gc = new GregorianCalendar(y, m, 1);
+        int daysInMonth = DAYS_IN_MONTH[m];
+        if (gc.isLeapYear(gc.get(Calendar.YEAR)) && m == 1) {
+            daysInMonth++;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (int i = 1; i <= daysInMonth; i++) {
+            sb.append(i).append(" ");
+        }
+        String[] allDays = sb.toString().split("\\s");
+        String[] neededDays = Arrays.copyOfRange(allDays, 0, tailGap);
+        return neededDays;
+    }
+
+    /**
+     * Combines 3 string arrays together, with array {@code a}'s elements coming in front, array
+     * {@code b}'s elements coming in the middle and array {@code c}'s elements coming at the back.
+     * @param a String array whose elements are to be in front
+     * @param b String array whose elements are to be in the middle
+     * @param c String array whose elements are to be behind
+     * @return String[] Combined String array.
+     */
+    private String[] combine(String[] a, String[] b, String[] c) {
```

Comment:
Perhaps can consider giving `a`, `b` and `c` more meaningful names like `previous month`, `current month` and `next month`.

Reply:
alright will make the changes!